### ISSUE TITLE
Cannot pass repeated CVararg params on to external function

### ISSUE THREAD

**Issue by jokade** [scala-native/scala-native#1142]
Is this somehow related to #396? I couldn't find an open issue for this (although I'd thought that's a quite common use case)...

```scala
def foo(args: CVararg*): Unit = Foo.bar(args:_*)

@extern
object Foo {
  def bar(args: CVararg*): Unit = extern
}
```

```
java.util.NoSuchElementException: None.get
	at scala.None$.get(Option.scala:347)
	at scala.None$.get(Option.scala:345)
	at scala.scalanative.nscplugin.NirGenExpr$ExprBuffer$$anonfun$genMethodArgs$2.apply(NirGenExpr.scala:1665)
```

**Comment by densh** [comment]
Finally found a design that works with this in #1426. Instead of compiling to C's varargs, we'll compile to `va_list`. Unlike the previous design it can be abstracted over and passed as normal value as requested in this issue.

**Comment by densh** [comment]
Fixed in #1426
